default/python/AI/ShipDesignAI.py

```diff
@@ -745,6 +746,7 @@ def update_stats(self, ignore_species: bool = False):
                 shield_counter += 1
                 if shield_counter == 1:
                     self.design_stats.shields = capacity
+                    self.design_stats.shield_type = part.name
                 else:
                     self.design_stats.shields = 0
```

Comment:
This code looks is odd, there should be no more than one shield, shouldn't it?
But the else case is actually triggered...